lightning/src/routing/scoring.rs

```diff
@@ -1218,7 +1218,7 @@ impl<L: DerefMut<Target = u64>, BRT: DerefMut<Target = HistoricalBucketRangeTrac
 
 	/// Adjusts the upper bound of the channel liquidity balance in this direction.
 	fn set_max_liquidity_msat(&mut self, amount_msat: u64) {
-		*self.max_liquidity_offset_msat = self.capacity_msat.checked_sub(amount_msat).unwrap_or(0);
+		*self.max_liquidity_offset_msat = self.available_capacity().saturating_sub(amount_msat);
```

Comment:
Hmmm.. I don't think this is correct. `max_liquidity_offset_msat` is an offset, so in-flight HTLCs shouldn't come into play. They'll be accounted for when calling `max_liquidity_msat`.

Reply:
Though perhaps we need to account for in-flight HTLCs for what we pass to `set_max_liquidity_msat`.

Reply:
If this is not a bug then I think `set_min_liquidity_msat` has a bug because `max_liquidity_msat` takes into account inflight htlcs